desk/app/channels-server.hoon

```diff
@@ -92,27 +92,149 @@
+      log      log
+      future   (future-6-to-7 future.v)
+    ==
+  ++  v-posts-6-to-7
+    |=  [vp=v-posts:v7:old:c mod=(map id-post:c time)]
+    ^-  [@ud v-posts:c]
+    =|  posts=v-posts:c
+    %-  (rep:mo-v-posts:v7:old:c vp)
+    |=  [[=id-post:c post=(unit v-post:v7:old:c)] count=@ud =_posts]
+    ^+  [count posts]
+    ::  for each post traversed, even if it was deleted,
+    ::  .count increases to generate correct post sequence
+    ::  number
+    ::
+    =.  count  +(count)
+    :-  count
+    ?~  post  posts
+    ::  insert seq and mod-at into seal
+    ::
+    =/  new-post=v-post:c
+      =/  new-seal=v-seal:c
+        =+  mod-at=(~(got by mod) id-post)
+        =*  seal  -.u.post
+        =+  seal(|1 [+(count) mod-at |1.seal])
+        %=  -
+          replies  (v-replies-6-to-7 replies.seal)
+          reacts   (v-reacts-6-to-7 reacts.seal)
+        ==
+      [new-seal +.u.post]
+    (put:on-v-posts:c posts id-post `new-post)
+  ::
+  ++  v-replies-6-to-7
+    |=  =v-replies:v7:old:c
+    ^-  v-replies:c
+    %+  run:on-v-replies:v7:old:c  v-replies
+    |=  v-reply=(unit v-reply:v7:old:c)
+    ^-  (unit v-reply:c)
+    ?~  v-reply  ~
+    %=  v-reply
+      reacts.u  (v-reacts-6-to-7 reacts.u.v-reply)
+    ==
+  ::
+  ++  v-reacts-6-to-7
+    |=  =v-reacts:v7:old:c
+    ^-  v-reacts:c
+    %-  ~(run by v-reacts)
+    |=  v-react=(rev:c (unit react:v7:old:c))
+    ^-  (rev:c (unit react:c))
+    ?~  +.v-react  [-.v-react ~]
+    =+  rat=(kill:em u.v-react)
+    ?~  rat
+      v-react(u [%any u.v-react])
+    v-react(u u.rat)
+  ::
+  ++  log-6-to-7
+    |=  l=log:v-channel:v7:old:c
+    ^-  [log:v-channel:c (map id-post:c @da)]
+    =|  seq-log=(map id-post:c @ud)
+    =|  =log:c
+    =|  mod=(map id-post:c @da)
+    =<  +
+    %-  (rep:mo-log:v7:old:c l)
+    |=  [[=time =u-channel:v7:old:c] [count=@ud =_seq-log] =_log =_mod]
+    ^+  [[count seq-log] log mod]
+    ?.  ?=(%post -.u-channel)
+      :-  [count seq-log]
+      :_  mod
+      (put:log-on:c log time u-channel)
+    ?.  ?=(%set -.u-post.u-channel)
+      :-  [count seq-log]
+      :_  (~(put by mod) id.u-channel time)
+      (put:log-on:c log time u-channel)
+    ?~  post.u-post.u-channel
+      :-  [count seq-log]
+      :_  (~(put by mod) id.u-channel time)
+      (put:log-on:c log time %post id.u-channel %set ~)
+    ::  increment .seq only for a new post
+    ::
+    =^  seq=@ud  count
+      ?~  seq=(~(get by seq-log) id.u-channel)
+        =.  count  +(count)
+        [count count]
+      [u.seq count]
+    =*  post  u.post.u-post.u-channel
+    =/  =u-post:c
+      :-  %set
+      (some post(|1.- [seq time |1.-.post]))
+    :-  :-  count
+        (~(put by seq-log) id.u-channel count)
+    :_  (~(put by mod) id.u-channel time)
+    (put:log-on:c log time %post id.u-channel u-post)
+  ::
+  ++  future-6-to-7
+    |=  f=future:v-channel:v7:old:c
+    ^-  future:v-channel:c
+    ::  channel future is defunct for now
+    ::
+    *future:v-channel:c
```

Comment:
Little bit unprincipled to nuke the data just so you don't have to write the migration for it. Would be good to write the conversion still, even if in the wild the future is always empty.